src/lib/clients/keeper.ts

```diff
@@ -1,43 +1,39 @@
-import { CommandClient, Handler } from "../../types/generic";
+import { CommandClient, Command } from '../../types/generic';
 import { KeeperClientOptions, KeeperSlackMiddleware } from '../../types/clients/keeper'
-import SlackClient from "./slack";
-import KeeperCommandHandlerMap from "../handlers/keeper";
-import { Logger } from "../../types/logger";
+import KeeperCommandHandlerMap from '../handlers/keeper';
+import SlackClient from './slack';
+import PhonyClient from './phony';
+import { Logger } from '../../types/logger';
 
 class KeeperClient implements CommandClient<KeeperClient> {
     private logger: Logger;
-    private slackClient: SlackClient;
-    private slackCommandHandlers: { [command: string]: Handler<KeeperSlackMiddleware> }
+    private slackClient: CommandClient<SlackClient | PhonyClient>;
```

Comment:
Nitpick (feel free to ignore): I wonder if we could just use `Client` here? That way we keep the interface generic. Or perhaps `slackClient` as the property name is too specific?

Reply:
I kinda like the `CommandClient` since it specified what we need to be consistent between `SlackClient` and `PhonyClient`. If it needs to be more generic in the future for any reason that would make sense but for now it seems better to enforce the availability of `registerCommand`.

As for `slackClient` as the property name -- the main reason I kept it is because the `PhonyClient` specifically is a phony _slack_ client. Maybe calling it `PhonySlackClient` is more specific? Thoughts?

Reply:
Sorry, I meant `CommandClient<Client>`, since they both are clients. I think a rename to `PhonySlackClient` is definitely better!

Reply:
Ah that makes sense! I started feeling it was strange that we needed to pass a type parameter there anyways, it looks like we can just set the return type on `registerCommand` to `CommandClient` and then we don't need the parameter at all!